### PR TITLE
bsunanda:Run2-hcx89 Add DB objects for HCAL TP 

### PR DESCRIPTION
**Pull request by bsunanda**
This adds 2 data base objects: HcalTPParameters which summarizes the TP implementation and HcalTPChannelParameters which contains parameters for each channel

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CalibFormats/HcalObjects/interface/HcalDbRecord.h [hunk 1]
```diff
@@ -28,10 +28,12 @@
 // class HcalDbRecord : public edm::eventsetup::EventSetupRecordImplementation<HcalDbRecord> {};
 
 class HcalDbRecord : public edm::eventsetup::DependentRecordImplementation <HcalDbRecord,  
-  boost::mpl::vector<HcalRecNumberingRecord, IdealGeometryRecord, HcalPedestalsRcd, HcalPedestalWidthsRcd, HcalGainsRcd, HcalGainWidthsRcd, 
+  boost::mpl::vector<HcalPedestalsRcd, HcalPedestalWidthsRcd, HcalGainsRcd, HcalGainWidthsRcd,
```
Comment (mmusich) [inline CalibFormats/HcalObjects/interface/HcalDbRecord.h]: @bsunanda why have you removed the dependency on `HcalRecNumberingRecord, IdealGeometryRecord`?

**Comment by bsunanda** [comment]
@mmusich I have a problem of going over 20 dependencies. I do not know how to get around that

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
please test

**Comment by bsunanda** [comment]
@mmusich @ggovi Please review and approve this. We need these conditions for 2017 simulation and reconstruction

**Comment by bsunanda** [comment]
@mmusich @ggovi Sorry to remind again. Please review and approve this. We need these conditions for 2017 simulation and reconstruction

**Comment by abdoulline** [comment]
@mmusich @ggovi
Marco and Giacomo, we really are in  a zeitnot  situation with 2017 updates..

**Comment by franzoni** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**